R/utilities-qualification-gof.R

```diff
@@ -0,0 +1,409 @@
+      outputMappings <- gofPlotGroup$OutputMappings
+
+      plotGOFMetadata$groups[[caption]] <- list()
+      plotGOFMetadata$groups[[caption]]$outputMappings <- list()
+      plotGOFMetadata$groups[[caption]]$symbol <- symbol
+
+
+      for (omap in seq_along(outputMappings)) {
+        outputMapping <- outputMappings[[omap]]
+
+        projectName <- outputMapping$Project
+        simulationName <- outputMapping$Simulation
+        outputPath <- outputMapping$Output
+        color <- outputMapping$Color
+
+        observedDataPathInSimulation <- outputMapping$Output
+        observedDataSet <- outputMapping$ObservedData
+        observedDataSetFilePath <- configurationPlan$observedDataSets[configurationPlan$observedDataSets$id == outputMapping$ObservedData, ]$path
+
+        simulationFile <- configurationPlan$getSimulationPath(
+          project = projectName,
+          simulation = simulationName
+        )
+
+        simulationResultsFile <- configurationPlan$getSimulationResultsPath(
+          project = projectName,
+          simulation = simulationName
+        )
+
+        simulation <- loadSimulation(simulationFile, loadFromCache = TRUE)
+        simulationResults <- importResultsFromCSV(simulation = simulation, filePaths = simulationResultsFile)
+
+        outputs <- lapply(simulation$outputSelections$allOutputs, function(output) {
+          Output$new(output$path)
+        })
+        names(outputs) <- lapply(simulation$outputSelections$allOutputs, function(output) {
+          output$path
+        })
+        output <- outputs[[outputPath]]
+        molWeight <- simulation$molWeightFor(outputPath)
+        simulationDimension <- getQuantity(path = outputPath, container = simulation)$dimension
+
+
+
+        # Setup simulations dataframe
+        simulatedDataStandardized <- data.frame(
+          Time = simulationResults$timeValues,
+          Concentration = simulationResults$getValuesByPath(path = outputPath, individualIds = 0)
+        )
+
+        # Setup observations dataframe
+        observedDataFileData <- read.csv(file.path(inputFolder, observedDataSetFilePath), check.names = FALSE, fileEncoding = "UTF-8-BOM")
+        observedDataFileMetaData <- parseObservationsDataFrame(observedDataFileData)
+        # If simulation in c("Amount","'"Concentration (molar)") and observations in c("Mass","'"Concentration (mass)"), convert observations to c("Amount","'"Concentration (molar)")
+        if (simulationDimension %in% c(ospDimensions$Amount, ospDimensions$`Concentration (molar)`)) {
+          observationsDimension <- massMoleConversion(ospsuite::getDimensionForUnit(observedDataFileMetaData$output$unit))
+        }
+        # Verify that simulations and observations have same dimensions
+        validateIsIncluded(values = observationsDimension, parentValues = simulationDimension, nullAllowed = FALSE)
+
+        observedDataStandardized <- observedDataFileData[, c(1, 2)]
```

Comment:
@Yuri05 ,
Just wondering, at some point, would there be an interest to include the uncertainty around observed data in the GOF merged plots ?